cpu/stm32/periph/flashpage.c

```diff
@@ -185,17 +219,35 @@ void flashpage_write(void *target_addr, const void *data, size_t len)
     stmclk_enable_hsi();
 #endif
 
+#ifdef FLASH_ACR_DCEN
+    if (FLASH->ACR & FLASH_ACR_DCEN) {
+        FLASH->ACR &= ~FLASH_ACR_DCEN;
+    }
+#endif
+#ifdef FLASH_ACR_ICEN
+    if (FLASH->ACR & FLASH_ACR_ICEN) {
+        FLASH->ACR &= ~FLASH_ACR_ICEN;
+    }
+#endif
+
     /* unlock the flash module */
     _unlock_flash();
 
+#if defined(CPU_FAM_STM32F2) || defined(CPU_FAM_STM32F4) || \
+    defined(CPU_FAM_STM32F7)
```

Comment:
```suggestion
#if defined(FLASH_CR_PSIZE_1)
```

Would this also work?

Reply:
I think so, I checked the vendor headers and I think it is only defined for the f2, f4 and f7. Changed it